### PR TITLE
Update DG for undo command

### PR DESCRIPTION
**Pull request by Cp-John**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -137,79 +137,85 @@ This section describes some noteworthy details on how certain features are imple
...
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will 
+not executes `previousAppointmentLists.push(new ArrayList<>(appointments.asUnmodifiableObservableList()))`, so the 
+appointment book state will not be saved into the `previousAppointmentLists`.
 
 </div>
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+Step 4. The user now decides that adding the appointment was a mistake, and decides to undo that action by 
+executing the `undo` command. The `undo` command will call `Model#undoAppointmentBook()`, which will shift
+the `currentAppointmentBookStatePointer` to the most recently saved state, pointing it to the previous appointment
```
Comment (candyhy) [inline docs/DeveloperGuide.md]: Should this be combined into one line?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -137,79 +137,85 @@ This section describes some noteworthy details on how certain features are imple
...
+the `currentAppointmentBookStatePointer` to the most recently saved state, pointing it to the previous appointment 
+book state, and restores the appointment book to that state.
 
-![UndoRedoState3](images/UndoRedoState3.png)
+![UndoState3](images/UndoState3.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index 0, pointing to the initial AddressBook state, then there are no previous AddressBook states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `previousAppointmentLists` is empty,
+then there are no previous AppointmentBook states to restore. The `undo` command uses `commandHistory.empty()` 
+to check if this is the case. If so, it will return an error to the user rather
```
Comment (candyhy) [inline docs/DeveloperGuide.md:173]: Should this be combined into one line? 120 characters per line does not apply to DG

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -137,79 +137,85 @@ This section describes some noteworthy details on how certain features are imple
...
 than attempting to perform the undo.
 
 </div>
 
-The following sequence diagram shows how the undo operation works:
+Step 5. The user then decides to execute the command `list`. Commands that do not modify the appointment book, 
+such as `list`, will usually not executes `previousAppointmentLists.push(new ArrayList<>(appointments.asUnmodifiableObservableList()))`,
+Thus, the `previousAppointmentLists` remains unchanged.
```
Comment (candyhy) [inline docs/DeveloperGuide.md]: Should this be combined into one line?

**Review by candyhy — COMMENTED** [review]
LGTM with some comments

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Cp-John) [reply docs/DeveloperGuide.md]: Fixed. Thanks!

**Review by candyhy — APPROVED** [review]
LGTM

**Pull request merged**